Tests/xcodeprojMeasurementsTests/MeasurementsTests.swift

```diff
@@ -0,0 +1,97 @@
+import Foundation
+import PathKit
+import SwiftShell
+@testable import xcodeproj
+import XCTest
+
+final class MeasurementsTests: XCTestCase {
+    
+    let tempDirectory = Path.temporary
+    var projectPath: Path?
+    var sourceRoot: Path?
+    
+    override func setUp() {
+        super.setUp()
+        try? tempDirectory.delete()
+        self.projectPath = tempDirectory + Path("Project")
+        self.sourceRoot = tempDirectory + Path("Sources")
+        let projectPath = self.projectPath!
+        let sourceRoot = self.sourceRoot!
+        let xcodeproj = try! generateXcodeProj(path: projectPath, sourceRoot: sourceRoot)
+        validate(xcodeproj: xcodeproj)
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        try? tempDirectory.delete()
+    }
+    
+    func test_measureProjectRead() {
+        let duration = try! measureDuration {
+            _ = try XcodeProj(path: self.projectPath!)
+        }
+        XCTAssertLessThan(duration, 3)
+    }
+    
+    func test_measureProjectWrite() {
+        let xcodeproj = try! XcodeProj(path: self.projectPath!)
+        let duration = try! measureDuration {
+            try xcodeproj.write(path: self.projectPath!)
+        }
+        XCTAssertLessThan(duration, 5)
+    }
+    
+    func test_measureObtainFilesFullPath() {
+        let xcodeproj = try! XcodeProj(path: self.projectPath!)
+        var pathList = [Path]()
+        var count = 0
+        let duration = try! measureDuration {
+            for project in xcodeproj.pbxproj.projects {
+                for target in project.targets {
+                    for buildPhase in target.buildPhases {
+                        let files = buildPhase.files
+                        for file in files {
+                            let fullPath = try file.file!.fullPath(
+                                sourceRoot: sourceRoot!
+                            )
+                            pathList.append(fullPath!)
+                            count += 1
+                        }
+                    }
+                }
+            }
+        }
+        XCTAssertLessThan(duration, 0.1)
+    }
+    
+    func test_measureProjGeneration() {
+        let duration = try! measureDuration {
+            try! generateXcodeProj(path: projectPath!, sourceRoot: sourceRoot!)
+        }
+        XCTAssertLessThan(duration, 60)
+    }
+    
+    private func validate(xcodeproj: XcodeProj) {
+        let pbxproj = xcodeproj.pbxproj
+        let settings = XcodeProjGenerator.GenerateSetting()
+        let project = pbxproj.projects.first!
+        XCTAssertEqual(project.targets.count, settings.targetCount)
+        let expectedGroupCount = 1701
```

Comment:
I set the values, but they can be calculated. Can I leave it like this or better fix it?